jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsResultSerializerFactory.java

```diff
@@ -0,0 +1,90 @@
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import reactor.core.publisher.Flux;
+import se.fortnox.reactivewizard.json.JsonSerializerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+import static reactor.core.publisher.Flux.just;
+
+public class JaxRsResultSerializerFactory {
+    private static final Class<byte[]>   BYTEARRAY_TYPE = byte[].class;
+    private static final Charset         charset        = StandardCharsets.UTF_8;
+
+    private final JsonSerializerFactory jsonSerializerFactory;
+
+    @Inject
+    public JaxRsResultSerializerFactory(JsonSerializerFactory jsonSerializerFactory) {
+        this.jsonSerializerFactory = jsonSerializerFactory;
+    }
+
+    /**
+     * Creates a non streaming result serializer
+     * @param type Content-Type
+     * @param dataCls Return type of the resource method
+     * @param isFlux True if it is a flux, false otherwise
+     * @param <T> Type of elements emitted by the publisher
+     * @return a function that serializes the elements emitted by a publisher to a Flux of byte arrays.
+     */
+    public <T> Function<Flux<T>, Flux<byte[]>> createNonStreamingSerializer(String type, Class<T> dataCls, boolean isFlux) {
+        if (type.equals(MediaType.APPLICATION_JSON)) {
+            if (!isFlux) {
+                return flux -> flux.map(jsonSerializerFactory.createByteSerializer(dataCls));
+            } else {
+                return flux -> flux.buffer().defaultIfEmpty(Collections.emptyList()).map(jsonSerializerFactory.createByteSerializer(new TypeReference<>() { }));
+            }
+        }
+        if (dataCls.equals(BYTEARRAY_TYPE)) {
+            return flux -> flux.cast(byte[].class);
+        }
+
+        if (isFlux) {
+            return flux -> flux.map(Object::toString).reduce(String::concat).map(String::getBytes).flux();
+        }
+        return flux -> flux.map(object -> object.toString().getBytes());
+    }
+
+    /**
+     * Creates a streaming result serializer.
+     * @param type Content-Type
+     * @param dataCls Return type of the resource method
+     * @param streamType Stream type
+     * @param isFlux True if it is a flux, false otherwise
+     * @param <T> Type of elements emitted by the publisher
+     * @return a function that serializes the elements emitted by a publisher to a Flux of byte arrays.
+     */
+    public <T> Function<Flux<T>, Flux<byte[]>> createStreamingSerializer(String type, Class<T> dataCls, Stream.Type streamType, boolean isFlux) {
+        if (type.equals(MediaType.APPLICATION_JSON)) {
+            var byteSerializer = jsonSerializerFactory.createByteSerializer(dataCls);
+            if (streamType == Stream.Type.CONCATENATED_JSON_OBJECTS || !isFlux) {
+                return flux -> flux.map(byteSerializer);
```

Comment:
It is a flux of course, but what does it contain?  Maybe its better to call it items? 

items -> items.map(byteSerilizer)